src/type_checkers/array_access.cr

```diff
@@ -0,0 +1,24 @@
+module Mint
+  class TypeChecker
+    # type_error NegatedExpressionNotBool
+
+    def check(node : Ast::ArrayAccess) : Type
+      lhs = node.lhs
+      index = node.index
+
+      case lhs
+      when Ast::ArrayLiteral
+        if lhs
+          first = lhs.items.first
+          if first
+            return Type.new("Maybe", [check first])
+          end
+        end
+      when Ast::Variable
```

Comment:
I would leave out the non `Ast::ArrayLiteral` cases for a future feature, we need to figure out a way to track literals through the program so we can check them here, for which I don't have a good idea right now.